### PR TITLE
[E-matija < T392] Implement fromLists for map module

### PR DESCRIPTION
**Pull request by mpintaric55334**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/map_module/algorithm/map.cpp [hunk 1]
```diff
@@ -1 +1,26 @@
...
+
+void Map::FromLists(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::List list1 = arguments[0].ValueList();  // lef this non const, because of the fix
+    mgp::List list2 = arguments[1].ValueList();
+
+    const size_t n = list1.Size();
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:11]: `const auto` you can use here

Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:11]: I would change this to `expected_list_size`. Name should be more descriptive

File: cpp/map_module/algorithm/map.cpp [hunk 2]
```diff
@@ -1 +1,26 @@
...
+    if (n != list2.Size() || n == 0) {
+      throw mgp::ValueException("Lists must be of same size and not null");
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:13]: and not empty?

File: cpp/map_module/algorithm/map.cpp [hunk 3]
```diff
@@ -1 +1,26 @@
...
+    }
+    mgp::Map result = mgp::Map();
+    for (size_t i = 0; i < n; i++) {
+      result.Update(std::move(list1[i].ValueString()), std::move(list2[i]));
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:17]: do you need to check that list1[i] is of String type

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few changes, but looks good

File: cpp/map_module/algorithm/map.cpp (hunk 3 above)
Comment (mpintaric55334) [reply cpp/map_module/algorithm/map.cpp:17]: changed it so procedure only accets list::type::string

**Review by antoniofilipovic — APPROVED** [review]
Looks good

**Pull request merged**